connectors/overview.mdx

```diff
@@ -112,6 +112,32 @@ fn my_to_json(f: f64) -> String {
 used for storing data in data lakes. Arroyo supports writing Parquet via the FileSystem
 sink. Refer for the [FileSystem sink docs](/connectors/filesystem) for details.
 
+## Framing
+
+The framing configuration for a source determines how messages read from the source
+are split into records for processing. By default, this is set to `none`, which
+means that the source will emit a single record for each message it reads. This is
+particularly useful for sources—like HTTP endpoints—that do not have their own
+framing protocol.
+
+Framing is configured via the `framing` option when creating a SQL table. Currently
+only `newline` is supported, which splits the input on newlines.
+
+You may also set `framing.newline.max_length` to a number of bytes to limit the
```

Comment:
I should've noticed this when reviewing, but I'm worried about how this might interact with non-ASCII strings. If there's some UTF-8 emoji that crosses this boundary, it'll fail on deserialization, right? Probably not a huge deal, as it is up to the operator to use this flag.

Reply:
all strings in rust are utf8

Reply:
I believe this should work correctly for UTF-8—0x0A is always '\n' so you can search for ascii characters byte-by-byte and not be misled by multi-byte characters.

Reply:
Oh — sorry, you're referring to the max_length specifically. Yes, that may corrupt a multi-byte sequence, but decoding will still work because we use `from_utf8_lossy`.